packages/core-data/src/entities.js

```diff
@@ -317,11 +339,15 @@ async function loadPostTypeEntities() {
 				},
 				applyChangesToDoc: ( doc, changes ) => {
 					const document = doc.getMap( 'document' );
+
 					Object.entries( changes ).forEach( ( [ key, value ] ) => {
 						if (
 							document.get( key ) !== value &&
```

Comment:
The problem is that now, the document will always be different for the "blocks" property. Meaning if a user changes the "title", the "blocks" will also be considered "modified" and it will send them over the wire unnecessarily. I think there's probably several small consequences for this.

Reply:
Because it's a new array? What if we memo it? Ensure that for the same array, we return the same "serialisable" array.

Reply:
The requirement in CRDT "maps" is to avoid calling `set` if there's no change. In this case, if the "blocks"  object is still the same that exists in the `document`, we should not call `set`. 

For now it's not sufficient to memoize just the attribute because it will create a new "blocks" property anyway so we'll endup calling the `set` function.

---
That said, right now we're not "decomposing" the blocks property in CRDT which we should be doing to allow parallel modification of blocks and merging the changes. And once we do that, it will actually become similar to "memoize" per attribute.

Reply:
> The requirement in CRDT "maps" is to avoid calling `set` if there's no change. In this case, if the "blocks" object is still the same that exists in the `document`, we should not call `set`.

But we're not checking right now if the blocks have changed?

I'm not sure I understand much of the rest, but I want to understand if there's going to be a fundamental problem with it. 😅 The last bit of your comment sounds like it may be ok.

Reply:
> But we're not checking right now if the blocks have changed?

we are here; `document.get( key ) !== value` in this line if the "blocks" key in the document contains the same value as the one that we receive from state, we don't call the `set` function.

With the changes in this PR, the "blocks" key in the document will always be different because it's created every time when setting it, so there's no way it could match the value we get from the state.

Reply:
Oh missed that sorry. I'll try to fix it.

Reply:
This should work, right? Or am I misunderstanding it? d042da7eaeaf82d3433fc5d30aaeac9ad685eb2d